parsec/core/fs/storage/local_database.py

```diff
@@ -129,52 +204,63 @@ async def open_cursor(self, commit: bool = True) -> AsyncIterator[Cursor]:
         # Lock the access to the connection object
         async with self._lock:
 
-            # Get a cursor
-            cursor = self._conn.cursor()
-            try:
+            # Check connection state
+            self._check_state()
 
-                # Execute SQL commands
-                yield cursor
+            # Close the local database if an operational error is detected
+            with self._manage_operational_error():
 
-                # Commit the transaction when finished
-                if commit and self._conn.in_transaction:
-                    await self._run_in_thread(self._conn.commit)
+                # Execute SQL commands
+                cursor = self._conn.cursor()
+                try:
+                    yield cursor
+                finally:
+                    cursor.close()
 
-            # Close cursor
-            finally:
-                cursor.close()
+            # Commit the transaction when finished
+            if commit and self._conn.in_transaction:
+                await self._commit()
 
     async def commit(self) -> None:
         # Lock the access to the connection object
         async with self._lock:
 
+            # Check connection state
+            self._check_state()
+
             # Run the commit
-            await self._run_in_thread(self._conn.commit)
+            await self._commit()
 
     # Vacuum
 
-    def get_disk_usage(self) -> int:
+    async def get_disk_usage(self) -> int:
         disk_usage = 0
         for suffix in (".sqlite", ".sqlite-wal", ".sqlite-shm"):
             try:
-                disk_usage += self.path.with_suffix(suffix).stat().st_size
+                stat = await self.path.with_suffix(suffix).stat()
             except OSError:
                 pass
+            else:
+                disk_usage += stat.st_size
         return disk_usage
 
     async def run_vacuum(self) -> None:
-        # Vacuum disabled
-        if self.vacuum_threshold is None:
-            return
 
         # Lock the access to the connection object
         async with self._lock:
 
-            # Flush to disk
-            await self._run_in_thread(self._conn.commit)
+            # Check connection state
+            self._check_state()
+
+            # Vacuum disabled
+            if self.vacuum_threshold is None:
```

Comment:
Why the vacuum_threshold check here ?

Reply:
I thought it would make more sense to always raise an `FSLocalStorageClosedError` if the local database is closed, even if vacuum is disabled. But I don't have a strong opinion about it.

Reply:
👍